### PR TITLE
1.10.beta4 compatiblity

### PR DESCRIPTION
**Pull request by taras**
- Made {{ember-list}} helper compatible with Ember 1.10.beta4
- Fixed tests to start runloop when creating views
- Fixed a content test expect passed model
- Upgraded Ember Bower dependency to 1.10.beta4

/cc @stefanpenner @lukemelia

### CODE REVIEW

File: packages/list-view/lib/helper.js [hunk 1]
```diff
@@ -1,41 +1,20 @@
 import EmberListView from './list_view';
 import EmberVirtualListView from './virtual_list_view';
 
-function createHelper (view, options) {
```
Comment (jasonmit) [inline packages/list-view/lib/helper.js]: Would be sweet to support both handlebars and htmlbars

File: packages/list-view/lib/helper.js [hunk 2]
```diff
@@ -1,41 +1,20 @@
...
-
-  if (!hash.content) {
-    hash.content = 'this';
-    types.content = 'ID';
-  }
-
-  for (var prop in hash) {
-    if (/-/.test(prop)) {
-      var camelized = Ember.String.camelize(prop);
-      hash[camelized] = hash[prop];
```
Comment (jasonmit) [inline packages/list-view/lib/helper.js]: Is this no longer possible with htmlbars?

File: packages/list-view/lib/main.js [hunk 3]
```diff
@@ -11,5 +11,5 @@ Ember.ListItemView         = ListItemView;
 Ember.ListView             = ListView;
 Ember.ListViewHelper       = ListViewHelper;
 
-Ember.Handlebars.registerHelper('ember-list', EmberList);
-Ember.Handlebars.registerHelper('ember-virtual-list', EmberVirtualList);
+Ember.HTMLBars.registerHelper('ember-list', EmberList);
```
Comment (rwjblue) [inline packages/list-view/lib/main.js]: This has been moved to be a private API (in https://github.com/emberjs/ember.js/pull/10379).

Comment (taras) [inline packages/list-view/lib/main.js]: What's the alternative to doing this?

File: packages/list-view/tests/controller_test.js [hunk 4]
```diff
@@ -43,31 +43,40 @@ module("Ember.ListView controllers", {
...
 test("parent controller", function() {
   var watWasCalled = false;
-  var controller = Ember.Controller.extend({
-    foo: 'bar',
-    actions: {
-      wat: function() {
-        watWasCalled = true;
+  var controller;
+
+  Ember.run(function(){
```
Comment (stefanpenner) [inline packages/list-view/tests/controller_test.js]: @rwjblue any ideas what changed that make `extend` require a run ?

File: packages/list-view/lib/helper.js (hunk 2 above)
Comment (taras) [inline packages/list-view/lib/helper.js]: It's not necessary to do most of what old helper was doing to arguments. Please, read my comment above.

File: packages/list-view/tests/list_view_test.js [hunk 5]
```diff
@@ -56,13 +58,15 @@ test("should render an empty view when there is no content", function() {
...
 
-  view = Ember.ListView.create({
-    content: content,
-    height: height,
-    rowHeight: rowHeight,
-    itemViewClass: itemViewClass,
-    emptyView: emptyView
-  });
+  Ember.run(function(){
+    view = Ember.ListView.create({
```
Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: @taras what is causing the run-loops here, it should be sync.

File: demos/index.html [hunk 6]
```diff
@@ -19,7 +19,7 @@
 
   <body>
     <script type="text/x-handlebars">
-      {{#ember-list items=controller height=300 width=500 row-height=100 element-width=100}}
+      {{#ember-list items=controller height=300 width=500 rowHeight=100 element-width=100}}
```
Comment (stefanpenner) [inline demos/index.html]: i think this should remain consistent, row-height element-width.

File: packages/list-view/tests/controller_test.js (hunk 4 above)
Comment (rwjblue) [inline packages/list-view/tests/controller_test.js]: My guess is that the `create` caused the requirement to wrap in a run-loop, not the `extend`.

File: packages/list-view/tests/list_view_test.js (hunk 5 above)
Comment (taras) [inline packages/list-view/tests/list_view_test.js]: `Ember.ListView.create` requires run loop. I've seen this before, I'm not sure why it worked before.

File: demos/index.html (hunk 6 above)
Comment (taras) [inline demos/index.html]: Ok, I will add it.

File: packages/list-view/tests/controller_test.js (hunk 4 above)
Comment (taras) [inline packages/list-view/tests/controller_test.js]: Yes.

**Comment by stefanpenner** [comment]
> Fixed tests to start runloop when creating views

This seems in-error. Or something I am not aware of changed in ember, and may have been a mistake.

I am very happy this is moving to compatibility, but it does appear this isn't backwards compatible. Either we need to figure out a strategy to support both, or we need a different release/versioning strategy.

Going forward, I would love for list-view to just be in lock-step with ember, so the compatibility can also remain ember-version locked.

**Comment by taras** [comment]
Dasherized arguments have been restored.

File: demos/index.html (hunk 6 above)
Comment (taras) [inline demos/index.html]: Done.

**Comment by taras** [comment]
@stefanpenner @jasonmit I made the helper compatible with both Ember 1.9.1 & Ember 1.10.

I don't know how to test for this.

File: packages/list-view/tests/list_view_test.js (hunk 5 above)
Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: this is a bug.

Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: looking now

Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: its not a bug in your code, but it is making several issues in list-view apparent. I am also surprised this wasn't an issue pre the latest beta.

I need to grab dinner and stuff, I'll try to look more closely afterwords 

TL;DR initial render should create views with the right context, rather then create them, then add the context.

Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: Unfortunately this doesn't appear related to your PR, rather just something we will need to fix sometime afterwards https://github.com/emberjs/list-view/issues/196

Comment (taras) [inline packages/list-view/tests/list_view_test.js]: What should we do with what we have here?

Comment (stefanpenner) [inline packages/list-view/tests/list_view_test.js]: keep it for now. The fix is likely some-time off.

**Comment by stefanpenner** [comment]
> I don't know how to test for this.

did u test manually between those versions?

**Comment by taras** [comment]
Yeah, I blew away dependencies, changed version in bower.json, ran install and then ran tests

**Comment by stefanpenner** [comment]
> Yeah, I blew away dependencies, changed version in bower.json, ran install and then ran tests

Awesome, that is good enough for now.

It's abit tricky to get tests to run across multiple version. But maybe something some future effort can make easier for ember-addons.

**Comment by taras** [comment]
@stefanpenner should I updated dependency to 1.10?

**Comment by jasonmit** [comment]
@taras yes please <3

**Comment by TrueNorth** [comment]
Thanks for this @taras! This lets me put multiple projects on 1.10!

**Comment by stefanpenner** [comment]
woohoo!
